doc/mgmt/Model-based-Replace-Operation-HLD.md

```diff
@@ -0,0 +1,917 @@
+- For terminal nodes not in the payload & which are children of complex nodes in payload, fields having default values are identified and added to the defaultValueMap.
+- If tables are owned by multiple nodes as well as a target tree, all table entries belonging to a target instance are optionally updated with the payload without changing the fields that do not belong to the target yang hierarchy.
+Since these entries cannot be identified through the partial replace condition available in common_app.go, we need to add then to the UPDATE subOperMap. Eg: DEVICE_METADATA shared between openconfig-system and control plane / host  applications.. Non table owners are identified through the table-owner annotation.
+- If payload has data belonging to subtree, that subtree will be invoked and that subtree should also take care of deleting appropriate data according to its payload.This will be marked by infra as visited and won't be traversed for DELETE.The same applies if PUT target is annotated with subtree.If there exists a nested subtree under a parent subtree and its  a)Data present in payload - infra will invoke subtree during REPLACE and subtree should also take care of deleting appropriate data according to its payload.Infra will mark it as visited and won't traverse it in DELETE flow.b) Data NOT present in payload - infra will invoke it in DELETE traversal.
+- REPLACE on leaf results in UPDATE and REPLACE on leaf-list results in swapping the contents of leaf-list (existing behaviours.)
+
+2) The yang hierarchy from the target URI will be traversed by the infra to identify the tables / instances that are not part of the payload and marked to be deleted.
+
+- The instances to be deleted will be identified by doing a GET like traversal of "config true" nodes and by the data(table/key/ownership/fields , through inheritable table relationship with its parent.) learnt from the annotations .
+- For non table owners the fields that are part of the table that belong to the yang hierarchy are marked for delete.
+- For table owners the complete instance/ table entry will be identified for delete.
+- Subtree callbacks along the traversal path will be invoked and result from them will be merged into resultMap(existing behaviour).Only subtrees which were NOT visited in REPLACE payload processing will be invoked.
+- If there exists a nested subtree and its contents were not in payload then only that subtree and not the parent will be invoked in DELETE.
+- All the tables / instances identified for delete will be added to the subOpMap and merged to the final result map before we start the DB operations.
+
+
+### 3.1.1 Changes required by the Transformer Infra
+
+1) For target ending in list-instance which doesn't exist and is not annotated with subtree, it will be created apart from what is translated from payload(REPLACE).NO DELETE traversal for child yang hierarchy of target since instance never existed.
```

Comment:
For this case - "list instance creation" if not exist, I hope default value of the leaf nodes also will be created by infra along with the instance creation? Please confirm

Reply:
Yes. As part of resource creation, a default value will be populated if a YANG node defines it.